File: Anton-Kraev/Term2/Block4/Bash/Bash/Commands/WcCommand.cs
namespace Bash.Commands
{
    public class WcCommand : ICommand
    {
        public string Name => "wc";

        public string Execute(string args)
        {
            var result = "";
            using (var reader = new StreamReader(args))
            {
                var text = reader.ReadToEnd();
                result += text.Length.ToString() + " bytes\n";
                result += (text.Count(x => x == '\n') + 1).ToString() + " lines\n";
                result += (text.Count(x => x == ' ' || x == '\n') + 1).ToString() + " words";
            }
            return result;
        }
    }
}

File: IvanShurenkov/Second Term/Block4/Bash/lib/bash/src/main/kotlin/lib/pipe/commands/WcCommand.kt
package lib.pipe.commands

import lib.pipe.Output
import java.io.File

class WcCommand(val name: String) : ICommand {
    override fun run(args: Array<String>, input: String?): Output {
        val output = Output(null)
        for (arg in args) {
            try {
                val text = File(arg).readText()
                val countLines = text.count { it == '\n' }
                val countWords = Regex("""\s+""").findAll(text.trim()).count() + 1
                val countBytes = text.toString().toByteArray().count()
                output.addOut("$countLines $countWords $countBytes $arg")
            } catch (e: Exception) {
                //print(e.message)
                output.addErr(e.message)
            }
        }
        return output
    }
}